version: '3.8'

services:
  app:
    container_name: app_go_women_center
    build: .
    env_file:
      - .env
    networks:
      - network_go_women_center
    ports:
      - 8080:8080
    volumes:
      - ./:/usr/src/app
    restart: on-failure
    depends_on:
      db:
        condition: service_healthy
  db:
    container_name: db_go_women_center
    image: mysql:8-debian
    environment:
      - MYSQL_ROOT_PASSWORD=${DB_PASSWORD}
      - MYSQL_DATABASE=${DB_NAME}
      - MYSQL_USER=${DB_USERNAME}
      - MYSQL_PASSWORD=${DB_PASSWORD}
    healthcheck:
      test:
        [
          'CMD',
          'mysqladmin',
          'ping',
          '-h',
          'localhost',
          '-u',
          '$$MYSQL_USER',
          '-p $$MYSQL_ROOT_PASSWORD',
        ]
      timeout: 20s
      retries: 10
    networks:
      - network_go_women_center
    ports:
      - 3306:3306
    volumes:
      - ./db/backup:/var/lib/mysql:rw
    cap_add:
      - SYS_NICE  # CAP_SYS_NICE

networks:
  network_go_women_center:
    driver: bridge